components/proxy/src/main/java/com/hotels/styx/admin/handlers/RoutingObjectHandler.java

```diff
@@ -68,10 +64,11 @@
     public RoutingObjectHandler(StyxObjectStore<RoutingObjectRecord> routeDatabase, RoutingObjectFactory.Context routingObjectFactoryContext) {
         urlRouter = new UrlPatternRouter.Builder()
                 .get("/admin/routing/objects", (request, context) -> {
-                    String output = routeDatabase.entrySet()
+                    List<StyxObjectDefinition> objects = routeDatabase.entrySet()
                             .stream()
-                            .map(entry -> serialise(entry.getKey(), entry.getValue()))
-                            .collect(joining("\n"));
+                            .map(entry -> createObjectDef(entry.getKey(), entry.getValue()))
+                            .collect(Collectors.toList());
+                    String output = serialise(objects);
```

Comment:
Opinionated: I would inline `objects` because it is only ever passed, unconditionally, as an argument to `objects`. Therefore it just adds noise. :-)

Reply:
Possibly below, too.